examples/example-hono-api/src/index.ts

```diff
@@ -0,0 +1,23 @@
+import {Hono} from 'hono';
+import {JwtEnv, jwt, requireScope} from './auth0';
+import { logger } from 'hono/logger'
+
+const app = new Hono<JwtEnv>();
+app.use(logger())
+
+app.get('/api/public', (c) => {
+    return c.text(`Hello world!`);
+})
+
+app.use('/api/private/*', jwt());
+
+app.get('/api/private/', (c) => {
+    return c.text(`hello ${c.get('user').sub}`);
+})
+
+app.get('/api/private/scope', requireScope('read:data'), (c) => {
+    return c.text(`hello ${c.get('user').sub}`);
+})
```

Comment:
Is there a specific reason to prefer splitting it in two functions:

- jwt
- requireScope

I feel like that is easy to make mistakes and forget to add jwt etc. Why not just have one, with an optional scope:

```suggestion

app.get('/api/private/', jwt(), (c) => {
    return c.text(`hello ${c.get('user').sub}`);
});

app.get('/api/private/scope', jwt({ scopes: ['read:data'] }), (c) => {
    return c.text(`hello ${c.get('user').sub}`);
});
```

But this will require internal changes as well, but it would align with our fastify-api SDK.